Point_set_3/include/CGAL/Point_set_3/IO.h

```diff
@@ -401,24 +399,33 @@ read_ply_point_set(
   if (comments != NULL)
     *comments = reader.comments();
 
-  filler.instantiate_properties (reader);
+  for (std::size_t i = 0; i < reader.number_of_elements(); ++ i)
+  {
+    internal::PLY::PLY_element& element = reader.element(i);
 
-  point_set.reserve (reader.m_nb_points);
-  
-  std::size_t points_read = 0;
+    if (element.name() == "vertex" || element.name() == "vertices")
+    {
+      point_set.reserve (element.number_of_items());
+      filler.instantiate_properties (element);
+    }
   
-  while (!(stream.eof()) && points_read < reader.m_nb_points)
+    for (std::size_t j = 0; j < element.number_of_items(); ++ j)
     {
-      for (std::size_t i = 0; i < reader.readers().size (); ++ i)
-        reader.readers()[i]->get (stream);
+      for (std::size_t k = 0; k < element.number_of_properties(); ++ k)
+      {
+        internal::PLY::PLY_read_number* property = element.property(k);
+        property->get (stream);
 
-      filler.process_line (reader);
-      
-      ++ points_read;
+        if (stream.eof())
+          return false;
```

Comment:
In that case you should set the `failbit` of the stream.

Reply:
Actually, I just realized that this is completely wrong (if the input only has points, the _end of files_ is reached at the last iteration, and it's a normal behaviour, we don't want to return false in that case). I'll fix it (and I think I'll use `stream.bad()` as I did in the mesh reader).

Reply:
Be careful with `good()` and `bad()`. Their meaning is not always what one expects. In particular `bad()` is not the opposition of `good()`.

What is more, after `stream >> something`, the success of the operation if neither `good()` nor `!bad()`, but it is `!fail()`.

Reply:
Alright, I knew about `bad()` not being the opposite of `good()` (I used to do `return !stream.bad()` at the very end of the function), not about the `fail()` part.

For the above case, I think the best solution would be to do, _before_ attempting to read a number, this:
```c++
        if (stream.eof())
        {
          stream.setstate(std::ios::failbit);
          return false;
        }
```
Do you agree?

Reply:
- Returning `!bad()` is incorrect, because the stream could in a state with only the `failbit` set.

- Notice that there are cases were the `eofbit` is not set when reaching the end of a stream. For example, when `std::getline` can read the final `\n` character, it does not set the `eofbit`. It is only the next reading that will set it (and probably the `failbit` at the same time). That is why it is rather tricky to test `eof()` before reading.

A useful documentation page: https://en.cppreference.com/w/cpp/io/ios_base/iostate

If you have extra questions, I suggest we discuss together before a screen.